### PR TITLE
Fixed issues with file dialogs not stacking

### PR DESCRIPTION
**Pull request by OsiJr**
[AB#874](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/874)
[AB#1008](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/1008)

### CODE REVIEW

File: src/imgui_ex/vcFileDialog.cpp [hunk 1]
```diff
@@ -2,23 +2,156 @@
...
+
+  pProgramState->fileDialog.ppExtensions = ppExtensions;
+  pProgramState->fileDialog.numExtensions = numExtensions;
+
+  pProgramState->fileDialog.onSelect = callback;
+
+  ImGui::OpenPopup(udTempStr("%s##embeddedFileDialog", pLabel));
+}
+
+void vcFileDialog_ShowModal(vcState *pProgramState)
```
Comment (OsiJr) [inline src/imgui_ex/vcFileDialog.cpp]: This was the code that lived in main before

File: src/main.cpp [hunk 2]
```diff
@@ -55,15 +55,14 @@
...
+# include <emscripten/threading.h>
+# include <emscripten/emscripten.h>
 #elif UDPLATFORM_WINDOWS
-#include <windows.h>
-#include <shobjidl.h> 
+# include <windows.h>
 #endif
 
-UDCOMPILEASSERT(VDK_MAJOR_VERSION == 0 && VDK_MINOR_VERSION == 5 && VDK_PATCH_VERSION == 0, "This version of VDK is not compatible");
+UDCOMPILEASSERT(VDK_MAJOR_VERSION == 0 && VDK_MINOR_VERSION == 5 && VDK_PATCH_VERSION == 1, "This version of VDK is not compatible");
```
Comment (OsiJr) [inline src/main.cpp]: Unrelated change

File: src/imgui_ex/vcFileDialog.h [hunk 3]
```diff
@@ -42,21 +43,12 @@ struct vcFileDialog
   vcFileDialogCallback onSelect;
 };
 
-void vcFileDialog_Show(vcFileDialog *pDialog, char *pPath, size_t pathLen, const char **ppExtensions, size_t numExtensions, vcFileDialogType dialogType, vcFileDialogCallback callback);
-
-template <size_t N> inline void vcFileDialog_Show(vcFileDialog *pDialog, char(&path)[N], const char **ppExtensions, size_t numExtensions, vcFileDialogType dialogType, vcFileDialogCallback callback)
```
Comment (samsinsane) [inline src/imgui_ex/vcFileDialog.h]: Why delete these two templates? It seems reasonable to me to have these two options?

Comment (OsiJr) [reply src/imgui_ex/vcFileDialog.h]: I'm trying to remove the function entirely in favour of using the vcIMSW file picker but the changes required to do that were beyond this PR
